code/game/machinery/deployable.dm

```diff
@@ -132,13 +132,48 @@ for reference:
 			return
 
 /obj/structure/barricade/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)//So bullets will fly over and stuff.
+
+	if(istype(mover,/obj/item/projectile))
+		return (check_cover(mover,target))
+
 	if(air_group || (height==0))
 		return 1
 	if(istype(mover) && mover.checkpass(PASSTABLE))
 		return 1
 	else
 		return 0
 
+/obj/structure/barricade/proc/check_cover(obj/item/projectile/P, turf/from)
+	var/turf/cover
+	cover = get_step(loc, get_dir(from, loc))
+	if(!cover)
+		return 1
+	if (get_dist(P.starting, loc) <= 1) //Cover won't help you if people are THIS close
+		return 1
+	if (get_turf(P.original) == cover)
+		var/valid = FALSE
+		var/distance = get_dist(P.last_interact,loc)
+		P.check_hit_zone(loc, distance)
+
+		var/targetzone = check_zone(P.def_zone)
+		if (targetzone in list(BP_R_LEG, BP_L_LEG, BP_GROIN))
```

Comment:
This doesnt cover the feet

Reply:
Eris doesn't have feet and hands. That's why.

Reply:
Wha. I thought Eris was old bay code. Bay code had hands/feet..